cgen error for `_ := Color.from_string('abc')` when the user already had a custom static function with that name defined

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.0 2fec2fa.c5e1c8e
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2023-08-17 17:42:23

vroot: OK, value: /v/vnew
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.41.0
Git vroot status: weekly.2023.33-5-gc5e1c8ee-dirty (1 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg z.v`
```v
enum Color { red green blue }
fn Color.from_string(x string) Color { return Color.red }
_ := Color.from_string('abc')

```

**What did you expect to see?**

a successfull compilation, or a checker error, that redefining from_string is not allowed (the first is preferable, but the second is probably easier/faster to do, and will have less side effects)

**What did you see instead?**
```
==================
/tmp/v_1000/z.15343589090157703517.tmp.c:1870: error: ';' expected (got ".")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1720,7 +1720,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 			g.write('panic_debug(${paline}, tos3("${pafile}"), tos3("${pamod}"), tos3("${pafn}"),  ')
 			g.call_args(node)
 			g.write(')')
-		} else if node.name.ends_with('__static__from_string') {
+		} else if node.name.ends_with('__static__from_string') && !g.table.known_fn(node.name) {
 			if node.name !in g.str_fn_names {
 				g.gen_enum_static_from_string(node.name)
 				g.str_fn_names << node.name

--- a/vlib/v/tests/enum_custom_static_from_string_test.v
+++ b/vlib/v/tests/enum_custom_static_from_string_test.v
@@ -1,0 +1,14 @@
+enum Color {
+	red
+	green
+	blue
+}
+
+fn Color.from_string(x string) Color {
+	return Color.red
+}
+
+fn test_enum_custom_static_from_string() {
+	ret := Color.from_string('abc')
+	assert ret == Color.red
+}

